src/components/v5/common/Table/Table.tsx

```diff
@@ -0,0 +1,145 @@
+          onDuplicateRow={() => {
+            const values = getValues().payments;
+            const selectedRow = values.find(
+              (item) => item.key === row.original.key,
+            );
+            if (selectedRow) {
+              append([
+                {
+                  ...selectedRow,
+                  key: uuidv4(),
+                },
+              ]);
+            }
+          }}
+        />
+      ),
+    }),
+  ];
+
+  const table = useReactTable({
+    data: fields,
+    columns: [...columns, ...burgerColumn],
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  return (
+    <div className={className}>
+      {tableTitle && <h5 className="text-2 mb-3">{tableTitle}</h5>}
+      {!!fields?.length && (
+        <div className="border border-gray-200 rounded-lg">
+          <table className="w-full">
+            <thead className="h-[2.1875rem] bg-gray-50 rounded-t-lg border-b border-gray-200">
+              {table.getHeaderGroups().map((headerGroup) => (
+                <tr key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => (
+                    <th
+                      key={header.id}
+                      className="w-1/3 text-left text-sm text-gray-600 font-normal rounded-t-lg px-4"
+                    >
+                      {header.isPlaceholder
+                        ? null
+                        : flexRender(
+                            header.column.columnDef.header,
+                            header.getContext(),
+                          )}
+                    </th>
+                  ))}
+                </tr>
+              ))}
+            </thead>
+            <tbody>
+              {table.getRowModel().rows.map((row) => {
+                const lastRow =
+                  table.getRowModel().rows[table.getRowModel().rows.length - 1];
+                return (
+                  <tr
+                    key={row.id}
+                    className={clsx('h-[3.125rem] relative', {
+                      'border-b border-b-gray-200': lastRow.id !== row.id,
+                    })}
+                    onClick={() => setSelectedRowId(row.id)}
+                  >
+                    {row.getVisibleCells().map((cell) => (
+                      <td key={cell.id} className="px-4 text-md text-gray-500">
+                        {flexRender(
+                          cell.column.columnDef.cell,
+                          cell.getContext(),
+                        )}
+                      </td>
+                    ))}
+                  </tr>
+                );
+              })}
+            </tbody>
+          </table>
+        </div>
+      )}
+      <Button
+        mode="primaryOutline"
+        iconName="plus"
+        size="small"
+        className="mt-6"
+        isFullSize={isMobile}
+        onClick={() => {
+          append(actionData);
+        }}
+      >
+        {actionText}
+      </Button>
```

Comment:
I see no reason to have this inside of a table component. Whenever you will have to do something different than a button you will probably have to add more and more props. Just add the button under the Table but outside of this component